containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -874,24 +874,18 @@ class UnorderedMap {
     if (m_hash_lists.data() != src.m_hash_lists.data()) {
       Kokkos::deep_copy(m_available_indexes, src.m_available_indexes);
 
-      using raw_deep_copy =
-          Kokkos::Impl::DeepCopy<typename device_type::memory_space,
-                                 typename SDevice::memory_space>;
-
-      raw_deep_copy(m_hash_lists.data(), src.m_hash_lists.data(),
-                    sizeof(size_type) * src.m_hash_lists.extent(0));
-      raw_deep_copy(m_next_index.data(), src.m_next_index.data(),
-                    sizeof(size_type) * src.m_next_index.extent(0));
-      raw_deep_copy(m_keys.data(), src.m_keys.data(),
-                    sizeof(key_type) * src.m_keys.extent(0));
+      // do the other deep copies asynchronously if possible
+      typename device_type::execution_space exec_space{};
+
+      Kokkos::deep_copy(exec_space, m_hash_lists, src.m_hash_lists);
+      Kokkos::deep_copy(exec_space, m_next_index, src.m_next_index);
+      Kokkos::deep_copy(exec_space, m_keys, src.m_keys);
       if (!is_set) {
-        raw_deep_copy(m_values.data(), src.m_values.data(),
-                      sizeof(impl_value_type) * src.m_values.extent(0));
+        Kokkos::deep_copy(exec_space, m_values, src.m_values);
       }
-      raw_deep_copy(m_scalars.data(), src.m_scalars.data(),
-                    sizeof(int) * num_scalars);
+      Kokkos::deep_copy(exec_space, m_scalars, src.m_scalars);
 
-      Kokkos::fence(
+      exec_space.fence(
```

Comment:
Why did you change to `exec_space`? Currently we synchronize everything (`Kokkos::fence()` calls `static_fence()` which ends up calling `cudaDeviceSynchronize`). With this change,  if you use `--default-stream per-thread` with the CUDA backend, you will only fence the default streams not the others, right?

Reply:
`deep_copy_view` does not take an execution space, so I would expect to have a global fence. Unless there is a good reason, I think we should revert this change.

Reply:
Ok, I'll revert this.